### PR TITLE
Bugfix: pending contacts shouldn't be available in contacts widget and viewcontacts

### PR DESCRIPTION
**Pull request by rabuzarus**
Foreign people shouldn't be allowed to see the users pending contacts.

This fixes https://github.com/friendica/friendica/issues/3155

### CODE REVIEW

**Comment by annando** [comment]
Aren't pending contacts always blocked? Have you checked if this pull request really fixes your issues?

File: include/api.php [hunk 1]
```diff
@@ -635,7 +635,7 @@ function api_get_user(App $a, $contact_id = Null, $type = "json"){
 
 		$r = q("SELECT count(*) as `count` FROM `contact`
 				WHERE  `uid` = %d AND `rel` IN ( %d, %d )
-				AND `self`=0 AND NOT `blocked` AND `hidden`=0",
+				AND `self`=0 AND NOT `blocked` AND NOT `pending` AND `hidden`=0",
```
Comment (Quix0r) [inline include/api.php]: `AND `self`=0 AND NOT `blocked` AND NOT `pending` AND `hidden`=0",` I see 2 different ways of handling "boolean" values (or flags).

File: include/api.php [hunk 2]
```diff
@@ -626,7 +626,7 @@ function api_get_user(App $a, $contact_id = Null, $type = "json"){
 		// count friends
 		$r = q("SELECT count(*) as `count` FROM `contact`
 				WHERE  `uid` = %d AND `rel` IN ( %d, %d )
-				AND `self`=0 AND NOT `blocked` AND `hidden`=0",
+				AND `self`=0 AND NOT `blocked` AND NOT `pending` AND `hidden`=0",
```
Comment (Quix0r) [inline include/api.php]: Two conflicting handling of "boolean" values in one line. Maybe decide for one here?

File: include/identity.php [hunk 3]
```diff
@@ -374,7 +374,10 @@ function profile_sidebar($profile, $block = 0) {
 			if (dbm::is_result($r))
 				$updated =  date("c", strtotime($r[0]['updated']));
 
-			$r = q("SELECT COUNT(*) AS `total` FROM `contact` WHERE `uid` = %d AND NOT `self` AND NOT `blocked` AND NOT `hidden` AND NOT `archive`
+			$r = q("SELECT COUNT(*) AS `total` FROM `contact`
```
Comment (Quix0r) [inline include/identity.php]: Nice to indent the SQL statement so it becomes more readable by humans (not maschines, of course).

**Comment by Quix0r** [comment]
@annando yes, let's hide pending "friend" requests, maybe some could be spam (massively sent out requests).

**Comment by rabuzarus** [comment]
yes I have checked it. It fixes it. You are right. I thought that pending contacts are always pending and blocked. But I have 2 gnu-social contacts (network "stat). I have rejected their connection requests and now they are with blocked=0 and pending=1 in the contact table.

You know I have investigated in the friendica connection process. But this was about friendica <-> friendica contacts. So I don't no why it is done in other ways for other networks (maybe because they can't authenticate).

But this doesn't influence this PR.
We maybe need to check what this does mean for https://github.com/friendica/friendica/pull/2952 and https://github.com/friendica/friendica/pull/2979

> I see 2 different ways of handling "boolean" values (or flags).

Yes but this query is commented out. So I just added the missing query condition but don't fix the coding style.

**Pull request merged**